File: src/main/templates/javascript_sdk/src/client/FlexmodelClientFactory.js
import { ApiClient } from './ApiClient.js';
import { FlexmodelClient } from './FlexmodelClient.js';

/**
 * Flexmodel客户端工厂
 * 提供便捷的客户端创建方法
 */
export class FlexmodelClientFactory {
  /**
   * 创建Flexmodel客户端
   * @param {string} baseURL 基础URL
   * @param {string} datasourceName 数据源名称
   * @param {Object} options 选项
   * @returns {FlexmodelClient} Flexmodel客户端
   */
  static createClient(baseURL, datasourceName, options = {}) {
    const apiClient = new ApiClient(baseURL, options);
    return new FlexmodelClient(apiClient, datasourceName);
  }

  /**
   * 创建Flexmodel客户端（带API Key认证）
   * @param {string} baseURL 基础URL
   * @param {string} datasourceName 数据源名称
   * @param {string} apiKey API密钥
   * @param {Object} options 其他选项
   * @returns {FlexmodelClient} Flexmodel客户端
   */
  static createClientWithApiKey(baseURL, datasourceName, apiKey, options = {}) {
    return this.createClient(baseURL, datasourceName, {
      ...options,
      apiKey
    });
  }

  /**
   * 创建Flexmodel客户端（带用户名密码认证）
   * @param {string} baseURL 基础URL
   * @param {string} datasourceName 数据源名称
   * @param {string} username 用户名
   * @param {string} password 密码
   * @param {Object} options 其他选项
   * @returns {FlexmodelClient} Flexmodel客户端
   */
  static createClientWithCredentials(baseURL, datasourceName, username, password, options = {}) {
    return this.createClient(baseURL, datasourceName, {
      ...options,
      username,
      password
    });
  }

  /**
   * 创建默认配置的客户端（用于开发环境）
   * @param {string} datasourceName 数据源名称
   * @param {Object} options 其他选项
   * @returns {FlexmodelClient} Flexmodel客户端
   */
  static createDefaultClient(datasourceName, options = {}) {
    return this.createClient('http://localhost:8080', datasourceName, options);
  }
}

File: src/main/templates/python_sdk/src/client/flexmodel_client_factory.py
"""
Flexmodel客户端工厂
"""

from typing import Dict, Any, Optional
from .api_client import ApiClient
from .flexmodel_client import FlexmodelClient


class FlexmodelClientFactory:
    """Flexmodel客户端工厂 - 提供便捷的客户端创建方法"""
    
    @staticmethod
    def create_client(
        base_url: str, 
        datasource_name: str, 
        options: Optional[Dict[str, Any]] = None
    ) -> FlexmodelClient:
        """
        创建Flexmodel客户端
        
        Args:
            base_url: 基础URL
            datasource_name: 数据源名称
            options: 配置选项
            
        Returns:
            Flexmodel客户端实例
        """
        api_client = ApiClient(base_url, options)
        return FlexmodelClient(api_client, datasource_name)
    
    @staticmethod
    def create_client_with_api_key(
        base_url: str,
        datasource_name: str,
        api_key: str,
        options: Optional[Dict[str, Any]] = None
    ) -> FlexmodelClient:
        """
        创建Flexmodel客户端（带API Key认证）
        
        Args:
            base_url: 基础URL
            datasource_name: 数据源名称
            api_key: API密钥
            options: 其他选项
            
        Returns:
            Flexmodel客户端实例
        """
        if options is None:
            options = {}
        options['api_key'] = api_key
        return FlexmodelClientFactory.create_client(base_url, datasource_name, options)
    
    @staticmethod
    def create_client_with_credentials(
        base_url: str,
        datasource_name: str,
        username: str,
        password: str,
        options: Optional[Dict[str, Any]] = None
    ) -> FlexmodelClient:
        """
        创建Flexmodel客户端（带用户名密码认证）
        
        Args:
            base_url: 基础URL
            datasource_name: 数据源名称
            username: 用户名
            password: 密码
            options: 其他选项
            
        Returns:
            Flexmodel客户端实例
        """
        if options is None:
            options = {}
        options['username'] = username
        options['password'] = password
        return FlexmodelClientFactory.create_client(base_url, datasource_name, options)
    
    @staticmethod
    def create_default_client(
        datasource_name: str,
        options: Optional[Dict[str, Any]] = None
    ) -> FlexmodelClient:
        """
        创建默认配置的客户端（用于开发环境）
        
        Args:
            datasource_name: 数据源名称
            options: 其他选项
            
        Returns:
            Flexmodel客户端实例
        """
        return FlexmodelClientFactory.create_client('http://localhost:8080', datasource_name, options)
